extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/UnionChunkWriter.java

```diff
@@ -0,0 +1,283 @@
+
+                if (mode == UnionChunkReader.Mode.Sparse) {
+                    innerSizedChunks[ii].ensureCapacity(chunk.size());
+                    innerSizedChunks[ii].get().fillWithNullValue(0, chunk.size());
+                } else {
+                    innerSizedChunks[ii].ensureCapacity(0);
+                }
+                innerChunks[ii] = innerSizedChunks[ii].get().asWritableObjectChunk();
+            }
+            for (int ii = 0; ii < chunk.size(); ++ii) {
+                final Object value = chunk.get(ii);
+                int jj;
+                for (jj = 0; jj < classMatchers.size(); ++jj) {
+                    if (value == null || classMatchers.get(jj).isAssignableFrom(value.getClass())) {
+                        columnOfInterest.set(ii, (byte) columnOfInterestMapping[jj]);
+                        if (mode == UnionChunkReader.Mode.Sparse) {
+                            innerChunks[jj].set(ii, value);
+                        } else {
+                            int size = innerChunks[jj].size();
+                            columnOffset.set(ii, size);
+                            if (innerChunks[jj].capacity() <= size) {
+                                int newSize = Math.max(16, size * 2);
+                                innerSizedChunks[jj].ensureCapacityPreserve(newSize);
+                                innerChunks[jj] = innerSizedChunks[jj].get().asWritableObjectChunk();
+                            }
+                            innerChunks[jj].add(value);
+                        }
+                        break;
+                    }
+                }
+
+                if (jj == classMatchers.size()) {
+                    if (value == null) {
+                        throw new UnsupportedOperationException("UnionChunkWriter found null-value without null child");
+                    }
+                    throw new UnsupportedOperationException("UnionChunkWriter found unexpected class: "
+                            + value.getClass() + " allowed classes: " +
+                            classMatchers.stream().map(Class::getSimpleName)
+                                    .collect(Collectors.joining(", ")));
+                }
+            }
+            innerColumns = new DrainableColumn[numColumns];
+            for (int ii = 0; ii < numColumns; ++ii) {
+                final ChunkType chunkType = writerChunkTypes.get(ii);
+                final ChunkWriter<Chunk<Values>> writer = writers.get(ii);
+                final WritableObjectChunk<Object, Values> innerChunk = innerChunks[ii];
+
+                if (classMatchers.get(ii) == Boolean.class) {
+                    // do a quick conversion to byte since the boolean unboxer expects bytes
+                    for (int jj = 0; jj < innerChunk.size(); ++jj) {
+                        innerChunk.set(jj, BooleanUtils.booleanAsByte((Boolean) innerChunk.get(jj)));
+                    }
```

Comment:
Not hit.

Reply:
This was because testBit exited too early in your run; I see it covered now.